src/Components/AddRoom/AddRoom.tsx

```diff
@@ -0,0 +1,221 @@
+                      InputLabelProps={{
+                        style: { color: 'black' },
+                      }}
+                      sx={{
+                        bgcolor: "#F7F7F7", mb: 5, display: "flex",
+                        justifyContent: "flex-end",
+                      }}
+                      variant='filled'
+                      label="Capacity"
+                      {...register("capacity",{required:true})}
+                    />
+                    {errors.capacity&&errors.capacity.type==="required"&& <Typography sx={{color:"red"}}>capacity is required</Typography>}
+
+                  </Grid>
+                  <Grid item xs={6}>
+                    <TextField
+                      InputLabelProps={{
+                        style: { color: 'black' },
+                      }}
+                      sx={{
+                        bgcolor: "#F7F7F7", mb: 5, display: "flex",
+                        justifyContent: "flex-start",
+                      }}
+                      variant='filled'
+                      label="Discount"
+                      {...register("discount",{required:true})}
+                    />
+                    {errors.discount&&errors.discount.type==="required"&& <Typography sx={{color:"red"}}>discount is required</Typography>}
+
+
+                  </Grid>
+                  <Grid item xs={6}>
+
+                    <Select
+                      options={facilities}
+                      onChange={(selectedValue) => setSelectedValue(selectedValue)}
+                      multi
+                    
+                    ></Select>
+                  </Grid>
+                </Grid>
+              </Box>
+              <Grid item xs={12} >
+                <Box
+                  sx={{
+                    textAlign:"center",
+                    border: "1px #009247 dashed",
+                    borderColor: "#009247",
+                    bgcolor:"#EAFAF1",
+                    borderRadius: 4,
+                    width:"100%",
+                    padding: "10px",
+                  }}
+                >
+                     <Button component="label"  startIcon={<FileUploadOutlinedIcon color='action' />}  onChange={handleImage} {...register('imgs')}>
+                     <VisuallyHiddenInput type="file" />
+                     </Button>
+                  <Typography sx={{color:'black'}}>Drag & Drop or <Box component="span" sx={{color:"#009247"}}> Choose a Room Images</Box>  to Upload</Typography>
+               
+                </Box>
+              </Grid>
+              <Grid item xs={12}   sx={{
+       marginLeft:"auto",
+       marginTop:3
+        
+        
+         
+        }}>
+    
+        <Button onClick={()=>navigate(-1)} type='button' variant="outlined" sx={{ color:'#203FC7',mr:5}}>
+          Cancel
+        </Button>
+        <Button onClick={()=>navigate(-1)} type='submit' variant="contained" sx={{ color:'white',bgcolor:'#203FC7'}}>
```

Comment:
The navigation should happen after a successful response from the backend. In the current implementation, if we left inputs empty, and clicked save, we will get directed to the rooms page.